src/frontend/src/binder/relation/mod.rs

```diff
@@ -43,20 +43,48 @@ pub enum Relation {
 }
 
 impl Binder {
-    /// return the (`schema_name`, `table_name`)
-    pub fn resolve_table_name(name: ObjectName) -> Result<(String, String)> {
-        let mut identifiers = name.0;
-        let table_name = identifiers
+    /// return first and second name in identifiers,
+    /// must have one name and can use default name as other one.
+    fn resolve_double_name(
+        mut identifiers: Vec<Ident>,
+        err_str: &str,
+        default_name: &str,
+    ) -> Result<(String, String)> {
+        let second_name = identifiers
             .pop()
-            .ok_or_else(|| ErrorCode::InternalError("empty table name".into()))?
+            .ok_or_else(|| ErrorCode::InternalError(err_str.into()))?
             .value;
 
-        let schema_name = identifiers
+        let first_name = identifiers
             .pop()
             .map(|ident| ident.value)
-            .unwrap_or_else(|| DEFAULT_SCHEMA_NAME.into());
+            .unwrap_or_else(|| default_name.into());
+
+        Ok((first_name, second_name))
+    }
+
+    /// return the (`schema_name`, `table_name`)
+    pub fn resolve_table_name(name: ObjectName) -> Result<(String, String)> {
+        Self::resolve_double_name(name.0, "empty table name", DEFAULT_SCHEMA_NAME)
+    }
+
+    /// return the ( `database_name`, `schema_name`)
+    pub fn resolve_schema_name(
+        default_db_name: &str,
+        name: ObjectName,
+    ) -> Result<(String, String)> {
+        Self::resolve_double_name(name.0, "empty schema name", default_db_name)
+    }
+
+    /// return the `database_name`
+    pub fn resolve_database_name(name: ObjectName) -> Result<String> {
+        let mut identifiers = name.0;
```

Comment:
You need to validate there must be only 1 argument. Otherwise if given "a.b.c", actually "a" will be used.